src/solver/norm/forward_layernorm2d_ck.cpp

```diff
@@ -0,0 +1,272 @@
+
+    return std::any_of(ln_ptrs.begin(), ln_ptrs.end(), [&ln_args](auto& ln_ptrs) {
+        return ln_args.IsSupportedBy(ln_ptrs);
+    });
+}
+
+template <typename LnPtrsType>
+typename LnPtrsType::iterator FindLnPtr(LnPtrsType& ln_ptrs,
+                                        const miopen::norm::ProblemDescription& problem)
+{
+    const auto ln_args = CKArgs{problem};
+    return std::find_if(ln_ptrs.begin(), ln_ptrs.end(), [&ln_args](auto& ln_ptrs) {
+        return ln_args.IsSupportedBy(ln_ptrs);
+    });
+}
+
+template <typename DeviceOpType,
+          typename CKArgsType,
+          typename CastType,
+          typename ProblemDescriptionType = ProblemDescription>
+ConvSolution MakeInvokerFactory([[maybe_unused]] const ExecutionContext& context,
+                                const miopen::norm::ProblemDescription& problem)
+{
+    auto ln_ptr      = DeviceOpType::GetInstances();
+    auto ln_ptr_iter = FindLnPtr(ln_ptr, problem);
+
+    if(ln_ptr_iter == ln_ptr.end())
+    {
+        MIOPEN_LOG_E("Layernorm kernel does not exist.");
+        return {miopenStatusInvalidValue};
+    }
+
+    ConvSolution result;
+    result.invoker_factory =
+        [ck_args   = CKArgsType{problem},
+         sh_ln_ptr = std::shared_ptr{std::move(*ln_ptr_iter)}](const std::vector<Kernel>&) mutable {
+            return [ck_args = std::move(ck_args), sh_ln_ptr = std::move(sh_ln_ptr)](
+                       const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+                const auto& data_ctx = primitive_parameters.CastTo<CastType>();
+                auto argument_ptr    = ck_args.MakeArgPtr(sh_ln_ptr,
+                                                       data_ctx.x,
+                                                       data_ctx.weight,
+                                                       data_ctx.bias,
+                                                       data_ctx.y,
+                                                       data_ctx.mean,
+                                                       data_ctx.rstd);
+                auto invoker_ptr     = sh_ln_ptr->MakeInvokerPointer();
+
+                const auto enable_profiling = handle.IsProfilingEnabled();
+                float elapsed_time =
+                    invoker_ptr->Run(argument_ptr.get(), {handle.GetStream(), enable_profiling});
+                if(enable_profiling)
+                {
+                    handle.ResetKernelTime();
+                    handle.AccumKernelTime(elapsed_time);
+                }
+            };
+        };
+    return result;
+}
+#endif
+
+bool Layernorm2DCKForward::IsApplicable(
+    [[maybe_unused]] const ExecutionContext& context,
+    [[maybe_unused]] const miopen::norm::ProblemDescription& problem) const
+{
+#if MIOPEN_USE_COMPOSABLEKERNEL
+    if(miopen::IsDisabled(MIOPEN_DEBUG_LAYERNORM2DCKFORWARD_CONV_CK_LN{}))
+        return false;
+    if(!problem.IsRank2Dim1())
+        return false;
+    if(!problem.IsLargeSize())
```

Comment:
And IIRC there is some member functions in tensor that may satisfy the solver needs. IIRC You can ask tensor
- Is it packed (and then return false if it is not)
- What is its flattened size (and return false if it exceeds some solver-specific limit)

Reply:
I added a few more conditions.

Reply:
[resolved]